chapters/packages.tex

```diff
@@ -326,3 +326,123 @@ \section{External resources}\label{external-resources}
+msgid "<descriptive text>"
+msgstr "<translation>"
+\end{lstlisting}
+\end{nonnormative}
+Only the keywords \lstinline!msgctxt!, \lstinline!msgid! and \lstinline!msgstr! are used.
+
+\begin{nonnormative}
+At first there can be an optional comment describing the location (file name and line number) of the text to translate.
+Multiple occurences of the same string can be listed here, separated by space.
+\end{nonnormative}
+
+The \lstinline!<context identifier>! following the keyword \lstinline!msgctxt! shall be the full name of the Modelica class (e.g., \lstinline!Modelica.Blocks.Math.Sin!) where the text appears.
+Short class definitions do not appear here.
+Texts in such classes belong to the enclosing full class definition.
+
+\begin{nonnormative}
+After the \lstinline!msgid! keyword the text string which shall be translated follows.
+\end{nonnormative}
+It should contain the original string from the Modelica text representation.
+\begin{nonnormative}
+Since in Modelica control sequences also start with a backslash and another backslash is used to use sequences literally or to hide double quotes, no change is required here.
+But Modelica allows strings to go over more than one line, gettext does not.
+Therefore, line breaks in Modelica must be encoded with \lstinline!"\n"! for gettext.
+
+In order to avoid long lines in the translated strings (following \lstinline!msgid! or \lstinline!msgstr!), strings may be split into smaller parts given on consecutive lines, e.g.
+\begin{lstlisting}
+"A
+B"
+\end{lstlisting}
+is converted to
+\begin{lstlisting}
+"A\n"
+"B"
+\end{lstlisting}
+These strings are concatenated when read.
+Please regard that if a \lstinline!msgid! string is given more than once in the same context, all occurrences are translated with the same (last) translation!
+
+The keyword \lstinline!msgstr! is followed by the translation of \lstinline!msgid!.
+With regard to control characters and line breaks, the same rules apply as for \lstinline!msgid!.
+In the template file this string is empty by definition.
+If this is empty in a language specific file the contents of \lstinline!msgid! may be used instead.
+\end{nonnormative}
+
+The texts in following Modelica constructs should be translated:
+\begin{itemize}
+\item description strings of component declarations and classes
+\item strings in the following annotations:
+\begin{itemize}
+\item \lstinline!Text.string!, \lstinline!Text.textString!
+\item \lstinline!missingInnerMessage!, \lstinline!obsolete!, \lstinline!unassignedMessage!
+\item \lstinline!Dialog.group!, \lstinline!Dialog.tab!
+\item \lstinline!Dialog.loadSelector.caption!, \lstinline!Dialog.loadSelector.filter!, \lstinline!Dialog.saveSelector.caption!, \lstinline!Dialog.saveSelector.filter!
+\item \lstinline!Documentation.info!, \lstinline!Documentation.revisions!
+\item \lstinline!Figure.title!, \lstinline!Plot.title!, \lstinline!Curve.legend!
+\end{itemize}
+\end{itemize}
+
+C-style and line comments are not translated.
```

Comment:
The terminology we use is this:
```suggestion
Delimited comments and rest-of-line comments are not translated.
```

But why talk about the two kinds of comments rather than just comments in general?
```suggestion
Comments are not translated.
```

Reply:
This terminology was used to make it clear that comments are meant and not the description strings of declarations which also serve often as comments that are presented at the GUI of the tool.
But if you think it is sufficiently clear, we can reduce it to the latter.

Reply:
My primary goal is to to avoid inconsistent use of terminology in the specification.  For maximum clarity, I'd therefore go with the former suggestion, but for brevity I could also see us going with the latter.